stl/inc/mdspan

```diff
@@ -525,38 +525,45 @@ public:
     constexpr mapping() noexcept               = default;
     constexpr mapping(const mapping&) noexcept = default;
 
-    template <class _OtherIndexType, enable_if_t<is_convertible_v<_OtherIndexType, index_type>, int> = 0>
-    constexpr mapping(const _Extents& _E_, const array<_OtherIndexType, _Extents::rank()>& _S_) noexcept : _Myext{_E_} {
+#ifndef __clang__ // TRANSITION, MSVC messes up CTAD when concepts are used here (needs further investigation)
+    template <class _OtherIndexType, enable_if_t<is_convertible_v<const _OtherIndexType&, index_type>
+                                                     && is_nothrow_constructible_v<index_type, const _OtherIndexType&>,
+                                         int> = 0>
+#else // ^^^ workaround / no workaround vvv
+    template <class _OtherIndexType>
+        requires is_convertible_v<const _OtherIndexType&, index_type>
+              && is_nothrow_constructible_v<index_type, const _OtherIndexType&>
+#endif // ^^^ no workaround ^^^
+    constexpr mapping(const _Extents& _E_, const span<_OtherIndexType, _Extents::rank()> _S_) noexcept : _Myext{_E_} {

```

Comment:
Pre-existing: `_E_` and `_S_` are macro-like identifiers (all caps and underscores) that we conventionally avoid (with a limited number of occurrences that we feel bad about but haven't bothered to change yet). These should gain one or more lowercase letters, e.g. `_Ex_` and `_Sp_` at a minimum.

Occurs below. Ok to clean up in a followup PR.